hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -186,7 +187,7 @@ private FileSlice getFileSliceFromOperation(CompactionOperation operation, Strin
     return new FileSlice(
         operation.getFileGroupId(),
         operation.getBaseInstantTime(),
-        baseFileOpt.isPresent() ? baseFileOpt.get() : null,
+        operationType == WriteOperationType.LOG_COMPACT || baseFileOpt.isEmpty() ? null : baseFileOpt.get(),
```

Comment:
Should the filtering or drop of base file happen inside `FileGroupReaderBasedAppendHandle` before being passed to the FGReader for compacting log files only so that the `fileSlice` in `FileGroupReaderBasedAppendHandle` still represents the latest file slice, to avoid confusion and possible breaking change during consolidation of append handles in the future?

Reply:
On the same topic of constructing `FileGroupReaderBasedAppendHandle`, the file slice is derived from the operation, which is also passed to `FileGroupReaderBasedAppendHandle`.  So should `FileGroupReaderBasedAppendHandle` take the `operation` only and construct the file slice within `FileGroupReaderBasedAppendHandle` instead of taking an additional argument `fileSlice`?  Similar for `FileGroupReaderBasedMergeHandle`.

Reply:
I've updated the code to avoid the confusion. The FGReader now has options to specify the base file, log files, and partition directly instead of constructing the file slice so I am using those now.